src/app/pages/storage/snapshots/snapshot-table/snapshot-list.component.ts

```diff
@@ -0,0 +1,327 @@
+import {
+  DialogService, ModalService, WebSocketService, AppLoaderService,
+} from 'app/services';
+import { CoreService } from 'app/services/core-service/core.service';
+import { SnapshotListEvent } from '../interfaces/snapshot-list-event.interface';
+import { snapshotChanged } from '../store/snapshot.actions';
+import { selectSnapshots, selectSnapshotState } from '../store/snapshot.selectors';
+
+@UntilDestroy()
+@Component({
+  templateUrl: './snapshot-list.component.html',
+  styleUrls: ['./snapshot-list.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [FormatDateTimePipe],
+})
+export class SnapshotListComponent implements OnInit, AfterViewInit {
+  error$ = this.store$.select(selectSnapshotState).pipe(map((state) => state.error));
+  isLoading$ = this.store$.select(selectSnapshotState).pipe(map((state) => state.isLoading));
+  isEmpty$ = this.store$.select(selectSnapshotsTotal).pipe(map((total) => total === 0));
+  @ViewChild(MatSort, { static: false }) sort: MatSort;
+  dataSource: MatTableDataSource<SnapshotListRow> = new MatTableDataSource([]);
+  defaultSort: Sort = { active: 'snapshot', direction: 'desc' };
+  filterString = '';
+  expandedRow: SnapshotListRow;
+  selection = new SelectionModel(true, []);
+  settingsEvent$: Subject<CoreEvent> = new Subject();
+  showExtraColumns$ = new BehaviorSubject(false);
+  toolbarConfig: ToolbarConfig;
+  emptyConfig: EmptyConfig = {
+    type: EmptyType.NoPageData,
+    title: this.translate.instant('No snapshots are available.'),
+    large: true,
+  };
+  loadingConfig: EmptyConfig = {
+    type: EmptyType.Loading,
+    large: false,
+    title: this.translate.instant('Loading...'),
+  };
+  errorConfig: EmptyConfig = {
+    type: EmptyType.Errors,
+    large: true,
+    title: this.translate.instant('Can not retrieve response'),
+  };
+
+  get displayedColumns(): string[] {
+    if (this.showExtraColumns$.value) {
+      return ['select', 'dataset', 'snapshot', 'used', 'created', 'referenced', 'actions'];
+    }
+    return ['select', 'dataset', 'snapshot', 'actions'];
+  }
+
+  constructor(
+    private dialogService: DialogService,
+    private ws: WebSocketService,
+    private translate: TranslateService,
+    private modalService: ModalService,
+    private cdr: ChangeDetectorRef,
+    private core: CoreService,
+    private loader: AppLoaderService,
+    private matDialog: MatDialog,
+    private store$: Store<SnapshotSlice>,
+  ) {
+    if (window.localStorage.getItem('snapshotXtraCols') === 'true') {
+      this.showExtraColumns$.next(true);
+    }
+  }
+
+  ngOnInit(): void {
+    this.setupToolbar();
+    this.getSnapshots();
+  }
+
+  ngAfterViewInit(): void {
+    this.store$.dispatch(loadSnapshots({ extra: this.showExtraColumns$.value }));
+    this.modalService.onClose$.pipe(untilDestroyed(this)).subscribe(() => {
+      this.getSnapshots();
```

Comment:
You are both manually reloading here and listening to updates in effects. Seems like one of the approaches should be used.